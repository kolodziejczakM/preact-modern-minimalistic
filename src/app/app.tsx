import { h } from 'preact';
import { Link, Route } from 'wouter-preact';
import { TypedComponent } from '../shared/typings/prop-types';
import { actions, StoreState } from '../store';
import { useAction, useSelector } from '@preact-hooks/unistore';
import { Home } from '../routes/home/home';
import { AboutMe } from '../routes/aboutMe/aboutMe';
import { Wrapper } from './app.styles';

interface ComponentProps {
    description: string;
}

export const App: TypedComponent<ComponentProps> = ({
    description,
}: ComponentProps) => {
    const increment = useAction(actions.increment);
    const count = useSelector((state: StoreState) => state.count);

    return (
        <Wrapper>
            <h1>{description}</h1>
            <header>
                <Link href="/">
                    <a>Home</a>
                </Link>
                <Link href="/about-me">
                    <a>About me</a>
                </Link>
            </header>
            <section>
                <h1>Testing store</h1>
                <section>
                    (see redux-dev-tools): <strong>{count}</strong>
                </section>
                <button onClick={increment}>Increment</button>
                <hr />
            </section>
            <main>
                <Route path="/" component={Home} />
                <Route path="/about-me" component={AboutMe} />
            </main>
        </Wrapper>
    );
};
