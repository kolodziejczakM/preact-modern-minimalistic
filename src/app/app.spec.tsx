import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render, fireEvent } from '@testing-library/preact';
import { StoreProvider } from '@preact-hooks/unistore';
import createStore from 'unistore';
import { App } from './app';

const storeMock = createStore({ count: 12 });

describe('App', () => {
    it('should render correctly', () => {
        const { asFragment } = render(
            <StoreProvider value={storeMock}>
                <App description="This is test description" />
            </StoreProvider>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    describe('increment button', () => {
        it('should increment displayed count value', () => {
            const { getByText } = render(
                <StoreProvider value={storeMock}>
                    <App description="This is test description" />
                </StoreProvider>
            );

            const incrementBtn = getByText('Increment');
            fireEvent.click(incrementBtn);

            expect(getByText('13')).toBeInTheDocument();
        });
    });
});
