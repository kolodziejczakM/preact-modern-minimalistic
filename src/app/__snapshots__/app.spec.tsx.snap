// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<DocumentFragment>
  <section
    class="0 "
  >
    <h1
      class="1 "
    >
      This is test description
    </h1>
    <header>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/about-me"
      >
        About me
      </a>
    </header>
    <section>
      <h1>
        Testing store
      </h1>
      <section>
        (see redux-dev-tools): 
        <strong>
          12
        </strong>
      </section>
      <button>
        Increment
      </button>
      <hr />
    </section>
    <main>
      <div>
        <section>
          <h1>
            Testing pre-rendering
          </h1>
        </section>
        <section>
          <button>
            Load more
          </button>
        </section>
        <div />
      </div>
    </main>
  </section>
</DocumentFragment>
`;
